### DISCUSSION TITLE
Is there a way to play animation / call event using code?

### DISCUSSION THREAD

**Discussion by AMAIOLAMO** [AvaloniaUI/Avalonia#6834] [Q&A] [answered]
as the Title says, I want a way to call some kind of animation / event:
for example:
I have
```xaml
<Style Selector="Button.Foo:something">
    <!-- insert animation here -->
</Style>
```

and in code:

```cs
 // ofc not the raise event here, but I need a similar one, if there is one tho
FooButton.RaiseEvent("something");
```

is there something similar to this? if there is, how should I use it, and is there any limitations?
if there's none, is there any workaround?
any help would be appreciated : )

**Comment by wieslawsoltes** [comment, marked as answer]
You need to define pseudo classes from code.

**Reply by wieslawsoltes** [reply]
> You need to define pseudo classes from code.

https://github.com/AvaloniaUI/Avalonia/blob/bbfb2c17d6ddc2b545e378e8b480fae0ac830707/src/Avalonia.Controls/Button.cs#L568

**Reply by AMAIOLAMO** [reply]
Thanks a lot! that's what I'm looking for
